From: Florian Albrechtskirchinger <falbrechtskirchinger@gmail.com>
Date: Fri, 10 Feb 2012 22:15:54 +0100
References: FATE#306586
Git-commit: 12fc9d0923ca70ae8960bccebac09d5c12f8c4d4
Subject: [PATCH] btrfs: honor umask when creating subvol root
Patch-mainline: v3.3-rc7

Set the subvol root inode permissions based on the current umask.
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/inode.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/fs/btrfs/inode.c b/fs/btrfs/inode.c
index bf392e5..6e0ee9b 100644
--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -6706,8 +6706,10 @@ int btrfs_create_subvol_root(struct btrfs_trans_handle *trans,
 	int err;
 	u64 index = 0;
 
-	inode = btrfs_new_inode(trans, new_root, NULL, "..", 2, new_dirid,
-				new_dirid, S_IFDIR | 0700, &index);
+	inode = btrfs_new_inode(trans, new_root, NULL, "..", 2,
+				new_dirid, new_dirid,
+				S_IFDIR | (~current_umask() & S_IRWXUGO),
+				&index);
 	if (IS_ERR(inode))
 		return PTR_ERR(inode);
 	inode->i_op = &btrfs_dir_inode_operations;
-- 
1.7.6.233.gd79bc

